titus-server-master/src/main/java/com/netflix/titus/master/mesos/kubeapiserver/model/v1/V1OpportunisticResource.java

```diff
@@ -137,4 +137,11 @@ public int getCpus() {
         }
         return spec.getCapacity().getCpu();
     }
+
+    public String getName() {
+        if (metadata == null || metadata.getName() == null) {
```

Comment:
Is this a valid state when any of these is null? Empty string seems misleading.

Reply:
You could use an unknown constant here like the other places you used one, but I am assuming this is not supposed to happen as the metadata name should be the unique key of the resource. Is that a correct assumption?

Reply:
that is correct, the name should never be null. Ideally I'd want this POJO to be immutable and require non-nulls, but GSON (used by kubernetes-client-java) doesn't support it, so I'm just being extra defensive here.